alfresco-health-processor-platform/src/main/java/eu/xenit/alfresco/processor/service/ProcessorAttributeService.java

```diff
@@ -0,0 +1,85 @@
+package eu.xenit.alfresco.processor.service;
+
+import lombok.AllArgsConstructor;
+import org.alfresco.service.cmr.attributes.AttributeService;
+import org.alfresco.service.cmr.attributes.DuplicateAttributeException;
+import org.alfresco.service.descriptor.DescriptorService;
+import org.alfresco.util.ParameterCheck;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+/**
+ * {@link AttributeService} wrapper that allows different Alfresco nodes to claim ranges of transactions.
+ */
+public class ProcessorAttributeService {
+
+    private static final Logger logger = LoggerFactory.getLogger(ProcessorAttributeService.class);
+
+    public static final String ATTR_KEY_HEALTH_PROCESSOR = "HealthProcessor";
+    public static final String ATTR_KEY_IS_RUNNING = "IsRunning";
+
+    protected final AttributeService attributeService;
+    protected final DescriptorService descriptorService;
+
+    public <T extends Serializable> T getAttribute(final String key, final T defaultValue) {
```

Comment:
I have tendency to implement such methods as generic not to have to fiddle with them later on.
Anything that is serialisable can be processed with this method.
If Alfresco accepts it, I'm fine with it :)

Reply:
The problem here is that you assume `T` to be the type of `defaultValue`. 
Suppose a `String` was persisted, this will explode if the `defaultValue` is e.g. a `Long`?

Reply:
public Serializable getAttribute(final String key, final Serializable defaultValue)

> Anything that is serialisable can be processed with this method.
Doesn't this apply to the above signature as well?

Reply:
Replaced everything with the actual type.